components/client/src/main/java/com/hotels/styx/client/connectionpool/SimpleConnectionPool.java

```diff
@@ -77,10 +77,12 @@ public Origin getOrigin() {
             Connection connection = dequeue();
             if (connection != null) {
                 borrowedCount.incrementAndGet();
+                sink.onCancel(() -> availableConnections.add(connection));
```

Comment:
Shouldn't we also reduce the borrowedCount as well?

Reply:
Yes, very true.  I will fix!

Reply:
Fixed